force-app/main/default/classes/Gifts.cls

```diff
@@ -93,6 +95,25 @@ public with sharing class Gifts {
         }
     }
 
+    @TestVisible
+    private void giftsToProcessingStatus() {
+        Boolean paymentIntegrationUnavailable =
+            !paymentIntegrationServiceConfig.isIntegrationEnabled();
+
+        for (DataImport__c gift : dataImports) {
+            Gift currentGift = new Gift(gift);
+
+            if (currentGift.hasAuthorizedPayment() && paymentIntegrationUnavailable) {
```

Comment:
No change needed right now, but this is an example where a GiftPayments domain would be helpful.  currentGift.hasAuthorizedPayment() should be Payment.isAuthorized() (or Payments.authorized()).  GiftPayments would be a collection of Payments that know their GiftId.